analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -0,0 +1,316 @@
+        Package(
+            id = extractPackageId(node, workingDir),
+            declaredLicenses = extractDeclaredLicenses(node),
+            description = extractPackageField(node, workingDir, "description"),
+            homepageUrl = node["homepage"].textValueOrEmpty(),
+            binaryArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+            sourceArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+            vcs = VcsInfo(
+                type = VcsType.GIT,
+                url = node["url"].textValueOrEmpty(),
+                revision = node["revision"].textValueOrEmpty()
+            )
+        )
+
+    private fun runInspectRawField(pkgName: String, workingDir: File, field: String): String {
+        return run(workingDir, "inspect", pkgName, "--raw", field).stdout
+    }
+
+    private fun removeProjectPackage(rootNode: JsonNode, definitionFile: File): List<JsonNode> {
+        val projectPackage: JsonNode? = rootNode.find {
+            // Contains because conanfile.py's reference string often includes other data.
+            it["reference"].textValueOrEmpty().contains(definitionFile.name)
+        }
+        return if (projectPackage != null) {
+            rootNode.minusElement(projectPackage)
+        } else {
+            rootNode.toList<JsonNode>()
+        }
+    }
+
+    private fun extractDeclaredLicenses(node: JsonNode): SortedSet<String> =
+        sortedSetOf<String>().apply {
+            val license = node["license"]
+            license?.forEach {
+                add(it.textValueOrEmpty())
+            }
+        }
+
+    private fun extractPackageId(node: JsonNode, workingDir: File) =
+        Identifier(
+            type = "Conan",
+            namespace = "",
+            name = extractPackageField(node, workingDir, "name"),
+            version = extractPackageField(node, workingDir, "version")
+        )
+
+    /**
+     * Runs 'conan inspect --raw' over a specified package to extract specified field.
+     */
+    private fun extractPackageField(node: JsonNode, workingDir: File, field: String): String {
+        return runInspectRawField(node["display_name"].textValueOrEmpty(), workingDir, field)
+    }
+
+    /**
+     * Project package options differ depending on which definition file is used:
+     * conanfile.py: allows us to run 'conan inspect conanfile.py' and extract useful project metadata
+     * conanfile.txt: does not allow 'conan inspect conanfile.txt'.
+     * TODO: Format of 'conan info' output of conanfile.txt may be such that we can get project metadata
+     *  from the 'requires' field... need to investigate whether this is a sure thing before implementing.
+     */
+    private fun extractProjectPackage(rootNode: JsonNode, definitionFile: File, workingDir: File): Package {
+        val projectPackageJson = rootNode.find {
+            it["reference"].textValueOrEmpty().contains(definitionFile.name)
+        }!!
```

Comment:
Better use `requireNotNull(projectPackageJson)` and eventually fail with a telling message rather than doing this case here.